### ISSUE TITLE
R: duckdb_register() should probably have an `overwrite` argument

### ISSUE THREAD

**Issue by DavisVaughan** [duckdb/duckdb#967]
To me it seems reasonable that if `dbWriteTable()` errors if a temp table already exists, then `duckdb_register()` should as well. I'd imagine that an `overwrite` argument with a default of `FALSE` to mirror `dbWriteTable()` might be useful here.

``` r
library(duckdb)

con_duck <- dbConnect(duckdb())

df <- data.frame(x = 1, y = 2)
df2 <- data.frame(a = 3, b = 4)

query <- "SELECT * FROM temp"

duckdb_register(con_duck, "temp", df)
dbGetQuery(con_duck, query)
#>   x y
#> 1 1 2

duckdb_register(con_duck, "temp", df2)
dbGetQuery(con_duck, query)
#>   a b
#> 1 3 4

dbWriteTable(con_duck, "temp", df)
#> Error in .local(conn, name, value, ...): Table temp already exists. Set overwrite=TRUE if you want
#>                   to remove the existing table. Set append=TRUE if you would like to add the new data to the
#>                   existing table.
```

<sup>Created on 2020-09-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

**Comment by hannesmuehleisen** [comment]
Yes, that would be sensible. Can you send a PR?

**Comment by Mytherin** [comment]
Implemented by #4850, exactly two years later :)
